File: src/test/java/problems/easy/MajorityElement/MajorityElementTest.scala
package problems.easy.MajorityElement

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.Arguments.of
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.Arguments.of
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream
import org.junit.jupiter.api.Assertions.assertEquals;
import org.junit.jupiter.params.provider.Arguments.of;
import io.qameta.allure.Epic;
import io.qameta.allure.Feature;
import io.qameta.allure.Story;
import io.qameta.allure.Epic;
import io.qameta.allure.Feature;
import io.qameta.allure.Story;
import java.util.stream.Stream;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;
import org.junit.jupiter.api.Tags;

@Epic("epic Problems")
@Feature("feature Easy")
@Story("story Majority element")
@DisplayName("suite Scala")
class MajorityElementTest {
  @ParameterizedTest
  @MethodSource(Array("getInputData"))
  @Tags(Array(Tag("tag Scala")))
  @DisplayName("test Majority element")
  def shouldCalculateMajorityElement(input: Array[Int], expectedResult: Int) = {
    assertEquals(expectedResult, SolutionScala.majorityElement(input))
  }
}

object MajorityElementTest {
  def getInputData(): Stream[Arguments] = {
    Stream.of(
      of(Array(3, 2, 3), 3),
      of(Array(2, 2, 1, 1, 1, 2, 2), 2)
    )
  }
}

File: src/main/java/problems/easy/MajorityElement/majority_element.test.js
const { majorityElement } = require("./majority_element");
const { assert, expect } = require("chai");
const { allure } = require("allure-mocha/runtime");

describe.each([
  [[3, 2, 3], 3],
  [[2, 2, 1, 1, 1, 2, 2], 2],
])("majorityElement(%o) should equal to %i", (inputArray, expectedResult) => {
  allure.epic("epic Problems");
  allure.feature("feature Easy");
  allure.story("story Majority element");
  allure.suite("suite JavaScript");
  allure.tag("tag JavaScript");
  test(`input = {inputArray}, returns {expectedResult}`, () => {
    allure.description("test Majority element");
    expect(majorityElement(inputArray)).toEqual(expectedResult);
  });
});
